src/components/QuicksightDashboard.tsx

```diff
@@ -0,0 +1,102 @@
+import * as React from "react";
+import { LibrariesData, LibraryData } from "../interfaces";
+import { Store } from "redux";
+import { RootState } from "../store";
+import { connect } from "react-redux";
+import ActionCreator from "../actions";
+
+export interface QuicksightDashboardStateProps {
+  isFetchingLibraries?: boolean;
+  libraries?: LibraryData[];
+}
+
+export interface QuicksightDashboardDispatchProps {
+  fetchLibraries?: () => Promise<LibrariesData>;
+}
+
+export interface QuicksightDashboardOwnProps {
+  store?: Store<RootState>;
+  dashboardId?: string;
+}
+
+export interface QuicksightDashboardProps
+  extends QuicksightDashboardStateProps,
+    QuicksightDashboardDispatchProps,
+    QuicksightDashboardOwnProps {}
+
+export interface QuicksightDashboardState {
+  embedUrl: string;
+}
+
+class QuicksightDashboard extends React.Component<
+  QuicksightDashboardProps,
+  QuicksightDashboardState
+> {
+  constructor(props) {
+    super(props);
+    this.state = { embedUrl: null };
+  }
+  componentDidMount() {
+    if (this.state.embedUrl) {
+      return;
+    }
+
+    this.props.fetchLibraries().then((libs) => {
+      const library_uuids: string = libs.libraries
+        .slice(0, Math.min(1, 10))// TODO: remove this once  https://github.com/ThePalaceProject/circulation/pull/1577 is merged
+        .map((l) => l.uuid)
+        .join(",");
+      try {
+        fetch(
+          "/admin/quicksight_embed/" +
+            this.props.dashboardId +
+            "?libraryUuids=" +
+            library_uuids
+        )
+          .then((response) => response.json())
+          .then((data) => this.setState({ embedUrl: data["embedUrl"] }))
+          .catch((error) => {
+            console.error(error);
+          });
```

Comment:
I changed this to dispatch to an action, but I don't think it is exactly right yet.  I would like to get this out since this is really just a rough pass.